src/disjoint_set.jl

```diff
@@ -126,6 +127,15 @@ mutable struct DisjointSets{T}
     end
 end
 
+DisjointSets() = DisjointSets{Any}()
+DisjointSets(T::Type) = DisjointSets{T}()
```

Comment:
Don't need this one.
These should all have been removed. this was a julia 0.3ish thing.
Before type-parameters were allowed in general circumstancess.

```suggestion
```